### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by j-vent**
Code using the authentication API which requires user login before processing post request. Also asks for nickname instead of just an email.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -35,11 +37,12 @@
 public class DataServlet extends HttpServlet {
   
   DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-
+  UserService userService = UserServiceFactory.getUserService();
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+       System.out.println("data doGet");
     int numComments = Integer.valueOf(request.getParameter("numComments"));
```
[the remark this answers is not in the document]
Comment (j-vent) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: Oh I see, I did not look into the differences between them, thanks!

File: portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    response.setContentType("text/html");
+    PrintWriter out = response.getWriter();
+    out.println("<h1>Set Nickname</h1>");
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String nickname = getUserNickname(userService.getCurrentUser().getUserId());
+      out.println("<p>Set your nickname here:</p>");
+      out.println("<form method=\"POST\" action=\"/nickname\">");
+      out.println("<input name=\"nickname\" value=\"" + nickname + "\" />");
```
[the remark this answers is not in the document]
Comment (j-vent) [reply portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java:46]: The example had their forms written inside Java so I assumed it was alright. Would redirecting to another html file be alright?

[the remark this answers is not in the document]
Comment (zstewar1) [reply portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java]: Yeah, adding another HTML page and building the form in JS is fine.

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet{
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+      boolean isLoggedIn;
+      String  url="";
```
Comment (landondare) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:34]: Small nit pick: add a space before and after equal signs. `url = ""`

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      UserService userService = UserServiceFactory.getUserService();
+      if(!userService.isUserLoggedIn()){
+        isLoggedIn =false;
```
Comment (landondare) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:37]: add a space after the equals sign. `= false`

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java (hunk 3 above)
Comment (j-vent) [reply portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:34]: ah sorry!

**Pull request merged**